### PR TITLE
Dev

### LINKED ISSUE #3: Global error handling
Catch all API errors (4XX and 5XX codes) and show a toast in the right top corner.
Take `message` field from the response for the toast.

Error response structure:
```json
{
  "code": "VALIDATION_ERROR",
  "message": "string",
  "timestamp": 1694977969853,
  "traceId": "string",
  "errors": [
    {
      "title": "string",
      "fieldPointer": "string",
      "fieldValue": {}
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by holus-bolus**
added the modal

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+.idea/
```
Comment (lipiridi) [inline .gitignore:43]: .idea folder was accidentally pushed to the branch

File: index.html [hunk 2]
```diff
@@ -0,0 +1,185 @@
+<!DOCTYPE html>
+<html lang='en'>
+<head>
+    <meta charset='UTF-8'>
+    <meta name='viewport' content='width=device-width, initial-scale=1.0'>
+    <title>Custom Dashboard</title>
+    <script src='https://cdn.jsdelivr.net/npm/chart.js'></script>
+    <link href='https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css' rel='stylesheet'>
```
Comment (lipiridi) [inline index.html]: Let's use the latest stable 5.3.3 version

File: scripts/apiErrorHandler.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  toastContainer.appendChild(toast);
+  setTimeout(() => {
+    toast.remove();
+  }, 5000);
+};
+
+
+
+ const handleApiErrors = async (response) => {
+  if (!response.ok) {
```
Comment (lipiridi) [inline scripts/apiErrorHandler.js]: It looks a bit more readable when if block is small.
My proposal is to flip the if condition, so code will look like:
```js
const handleApiErrors = async (response) => {
        if (response.ok) {
            return response.json();
        }

        const errorResponse = await response.json();
        const errorMessage = errorResponse?.message || 'An unknown error occurred';

        // Display the error message in the toast
        showToast(errorMessage);

        throw new Error(errorMessage);
    }
```

**Comment by holus-bolus** [comment]
Fixed the issues

**Review by lipiridi — APPROVED** [review]

**Pull request merged**